configure.ac

```diff
@@ -1705,44 +1705,62 @@ else
   if test -n "$GUILE_CONFIG" ; then
     if test x"$GUILE" = xyes; then
       AC_MSG_CHECKING([for guile executable])
-      if $GUILE_CONFIG info guile > /dev/null 2>&1; then
-        GUILE="`$GUILE_CONFIG info guile`"
+      # Try extracting it via guile-config first. If it's defined there it's the most reliable result
+      GUILE="`$GUILE_CONFIG info guile 2>/dev/null`"
+      if test -n "$GUILE";  then
         AC_MSG_RESULT([$GUILE])
       else
-        AC_MSG_RESULT([not in guile-config, constructing path])
+        AC_MSG_RESULT([not found via guile-config - constructing path])
         AC_MSG_CHECKING([for guile bindir])
         guile_bindir="`$GUILE_CONFIG info bindir`"
         AC_MSG_RESULT([$guile_bindir])
-        GUILE=$guile_bindir/guile
-        if ! test -f "$GUILE" ; then
-          GUILE=
-          AC_PATH_PROG(GUILE, guile)
-        fi
+        GUILE="$guile_bindir/guile"
+      fi
+      if ! test -f "$GUILE" ; then
+        GUILE=
+        AC_PATH_PROG(GUILE, guile)
       fi
     fi
 
-    if test -f "$GUILE" ; then
+    if ! test -f "$GUILE" ; then
+      AC_MSG_WARN([no suitable guile executable found. Disabling Guile])
```

Comment:
It seems that `$GUILE` being empty means it's disabled, which matches how other similar variables work and generally seems to make sense.

However we also seem to have a second disabled state where `$GUILE` is non-empty but not a valid file path.  I think it would be cleaner to set `GUILE=` here and then replace `test -f "$GUILE"` below with `test -z "$GUILE"`.

Reply:
Although I believe both ways of testing are correct I agree with your observation. Using the same test all the time makes the code much more readable.